File: app/src/main/java/tech/salroid/filmy/activities/Rating.java
package tech.salroid.filmy.activities;

import android.content.Context;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import tech.salroid.filmy.BuildConfig;
import tech.salroid.filmy.network_stuff.VolleySingleton;

/*
 * Filmy Application for Android
 * Copyright (c) 2016 Sajal Gupta (http://github.com/salroid).
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

class Rating {

    private static String imdb_rating = "0", tomatometer_rating = "0", audience_rating = "0", metascore_rating = "0", image = null, rottenTomatoPage = null;
    private static String OMDB_API_KEY = BuildConfig.OMDB_API_KEY;

    static void getRating(final Context context, String movie_id_final) {

        VolleySingleton volleySingleton = VolleySingleton.getInstance();
        RequestQueue requestQueue = volleySingleton.getRequestQueue();


        String BASE_RATING_URL = "http://www.omdbapi.com/?i=" + movie_id_final + "&apikey=" + OMDB_API_KEY + "&tomatoes=true&r=json";

        JsonObjectRequest jsonObjectRequestForMovieDetails = new JsonObjectRequest(BASE_RATING_URL, null,
                new Response.Listener<JSONObject>() {
                    @Override
                    public void onResponse(JSONObject response) {

                        try {

                            Boolean responseBool = response.getBoolean("Response");

                            if (responseBool){

                                imdb_rating = response.getString("imdbRating");
                                tomatometer_rating = response.getString("tomatoRating");
                                audience_rating = response.getString("tomatoUserRating");
                                metascore_rating = response.getString("Metascore");
                                image = response.getString("tomatoImage");
                                rottenTomatoPage = response.getString("tomatoURL");


                                //Above tomatometer does not work this does
                                JSONArray jsonArray = response.getJSONArray("Ratings");
                                for (int i = 0; i < jsonArray.length(); i++) {
                                    if (jsonArray.getJSONObject(i).getString("Source").equals("Rotten Tomatoes")) {
                                        tomatometer_rating = jsonArray.getJSONObject(i).getString("Value");
                                    }
                                }

                                setRatingCallback(context, imdb_rating, tomatometer_rating, audience_rating, metascore_rating, rottenTomatoPage);

                            }else
                                setRatingFailCallback(context);

                        } catch (JSONException e) {
                            e.printStackTrace();
                            setRatingFailCallback(context);
                        }

                    }


                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {

                // Log.e("webi", "Volley Error: " + error.getCause());
                setRatingFailCallback(context);
            }
        }
        );

        requestQueue.add(jsonObjectRequestForMovieDetails);

    }

    private static void setRatingFailCallback(Context context) {
        ((MovieDetailsActivity) context).setRatingGone();
    }


    private static void setRatingCallback(Context context, String imdb_rating, String tomatometer_rating, String audience_rating, String metascore_rating, String rottenTomatoPage) {
        ((MovieDetailsActivity) context).setRating(imdb_rating, tomatometer_rating, audience_rating, metascore_rating, rottenTomatoPage );
    }

}

File: app/src/main/java/tech/salroid/filmy/activities/Rating.kt
package tech.salroid.filmy.activities

import android.content.Context
import tech.salroid.filmy.network_stuff.VolleySingleton
import com.android.volley.toolbox.JsonObjectRequest
import org.json.JSONException
import tech.salroid.filmy.BuildConfig

internal object Rating {

    private var imdbRating = "0"
    private var tomatoMeterRating = "0"
    private var audienceRating = "0"
    private var metaScoreRating = "0"
    private var image: String? = null
    private var rottenTomatoPage: String? = null
    private const val OMDB_API_KEY = BuildConfig.OMDB_API_KEY

    @JvmStatic
    fun getRating(context: Context, movie_id_final: String) {

        val volleySingleton = VolleySingleton.getInstance()
        val requestQueue = volleySingleton.requestQueue
        val baseRatingUrl = "http://www.omdbapi.com/?i=$movie_id_final&apikey=$OMDB_API_KEY&tomatoes=true&r=json"

        val jsonObjectRequestForMovieDetails = JsonObjectRequest(baseRatingUrl, null,
                { response ->
                    try {
                        val responseBool = response.getBoolean("Response")
                        if (responseBool) {
                            imdbRating = response.getString("imdbRating")
                            tomatoMeterRating = response.getString("tomatoRating")
                            audienceRating = response.getString("tomatoUserRating")
                            metaScoreRating = response.getString("Metascore")
                            image = response.getString("tomatoImage")
                            rottenTomatoPage = response.getString("tomatoURL")

                            // Above TomatoMeter does not work this does
                            val jsonArray = response.getJSONArray("Ratings")
                            for (i in 0 until jsonArray.length()) {
                                if (jsonArray.getJSONObject(i).getString("Source") == "Rotten Tomatoes") {
                                    tomatoMeterRating = jsonArray.getJSONObject(i).getString("Value")
                                }
                            }

                            setRatingCallback(context, imdbRating, tomatoMeterRating, audienceRating, metaScoreRating, rottenTomatoPage)

                        } else setRatingFailCallback(context)
                    } catch (e: JSONException) {
                        e.printStackTrace()
                        setRatingFailCallback(context)
                    }
                }
        ) {
            setRatingFailCallback(context)
        }
        requestQueue.add(jsonObjectRequestForMovieDetails)
    }

    private fun setRatingFailCallback(context: Context) {
        (context as MovieDetailsActivity).setRatingGone()
    }

    private fun setRatingCallback(context: Context, imdb_rating: String, tomatoMeterRating: String, audienceRating: String, metaScoreRating: String, rottenTomatoPage: String?) {
        (context as MovieDetailsActivity).setRating(imdb_rating, tomatoMeterRating, audienceRating, metaScoreRating, rottenTomatoPage)
    }
}